# 4.The digital root of a number n is obtained as follows: Add up the digits n to get a new number. Add up the digits of that to get another new number. Keep doing this until you get a number that has only one digit. That number is the digital root. For example, if n = 45893 , we add up the digits to get 4 + 5 + 8 + 9 + 3 = 29 . We then add up the digits of 29 to get 2 + 9 = 11 . We then add up the digits of 11 to get 1 + 1 = 2 . Since 2 has only one digit, 2 is our digital root. Write a function that returns the digital root of an integer n . [Note: there is a shortcut, where the digital root is equal to n mod 9, but do not use that here.]

In [19]:
def digi_root(): # function name
    
    n = input("Enter an integer: ") # takes input of the integer 
    sum = 0
    while(n > 0 or sum > 9): # to find digital root, input must be greater than zero 
                             # and the sum of the digits must be greater than 9
            
        if(n == 0): # if input is 0, sum will be 0
            n = sum
            sum = 0
        # if input is grater than 0
        else:
            sum = sum + (n % 10)
            n = n / 10
    
    print "Digital Root is: ", sum

digi_root() # calling function

Enter an integer: 25371
Digital Root is:  9


Description:

Here n = 25371 is the given integer. First, we add digits of this integer(2+5+3+7+1 = 18). Then, each of the digits of 18 is added (1+8 = 9). Then, the outcome 9 is a single digit. So, it is the digital root.

Calculation:

sum = (25371 % 10) = 1

n = (25371 / 10) = 2537

sum = 1 + (2537 % 10) = 8

n = (2537 / 10) = 253

sum = 8 + (253 % 10) = 11

n = (253 / 10) = 25

sum = 11 + (25 % 10) = 16

n = (25 / 10) = 2

sum = 16 + (2 % 10) = 18

n = (2 / 10) = 0

sum = 18 + (0 % 10) = 18

n = (0 / 10) = 0

Therefore,the Sum of Digits is 2+5+3+7+1 = 18 then 1+8 = 9. Here 9 is a single digit that is lower than 10, so the function stops calling and returns the output which is 9.